File: longest-common-prefix/longest-common-prefix_.cpp
#include <string>

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string out = "";

        int smallest = strs[0].size(); 
        for ( int i = 1; i < strs.size(); i++ ) {
            if ( smallest > strs[i].size() ) {
                smallest = strs[i].size();
            }
        }

        char c;
        for(int n = 0; n < smallest; n++){
            c = strs[0][n]; 
            for(int i = 1; i<strs.size(); i++){
                if (c != strs[i][n]){
                    return out;
                }
            }
            out += c;
        }
        return out;
    }
};

File: longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        out = ""
        for i in range(min(map(len, strs))):
            if all([j[i] == strs[0][i] for j in strs]):
                out += strs[0][i]
            else:
                break
        return out